### ISSUE TITLE
Group (Autograder) Annotations by Type

### ISSUE THREAD

**Issue by Luro02** [kit-sdq/intelligrade#16]
There can be a lot of annotations and it becomes harder to have an overview/give sensible feedback.

The current Autograder implementation reduces the annotations before emitting it to the grading tool. Because of this the not emitted annotations will not be looked over by the grading tool user.

It seems sensible to have some sort of group like a collapsible and make intelligrade merge the final annotations (alternatively this could be done by Artemis4J)

**Comment by dfuchss** [comment]
@Luro02 this is fixed now with https://github.com/kit-sdq/artemis4j/pull/117 , right :) ?

**Comment by Luro02** [comment]
> @Luro02 this is fixed now with [kit-sdq/artemis4j#117](https://github.com/kit-sdq/artemis4j/pull/117) , right :) ?

No, this is something different. Artemis4J groups all annotations for the student, but one of the reasons why I implemented the merging in the first place (in the autograder), was that tutors were overwhelmed with the number of annotations in the annotation list.

Merging them before submitting, is a bad solution, because of how much work it would be to figure out false-positives/remove them from the merged annotation.

This issue suggests adding a collapsible entry in the annotation list, that groups the annotations (like artemis4j does).

I am thinking of something like this:
```
v Custom Comment | L5, L10, ... |               | AUTOGRADER | Here the merged message, other problems in (L10, L11, ...)
Custom Comment   | L5           | src/Main.java | AUTOGRADER | Some message
Custom Comment   | L10          | src/Main.java | AUTOGRADER | Some message
Custom Comment   | L11          | src/Main.java | AUTOGRADER | Some message
```
Where you can press on the row with a "v" to show all messages that are merged with it. By collapsible rows, I am thinking of something like this:
https://stackoverflow.com/a/9517974/7766117

**Comment by Luro02** [comment]
I was planning to investigate this myself, but I haven't had the time to do this yet. This issue will likely need another Artemis4J PR to expose some of the stuff that I implemented in the merging PR
